src/lib/work_partitioner/pending_zkapp_command.mli

```diff
@@ -0,0 +1,23 @@
+open Core_kernel
+open Snark_work_lib
+
+type t
+
+val create :
+     job:(Spec.Single.t, Id.Single.t) With_job_meta.t
+  -> unscheduled_segments:Spec.Sub_zkapp.Stable.Latest.t Base.Queue.t
+  -> pending_mergeable_proofs:Ledger_proof.t Deque.t
+  -> t
+
+val generate_job_spec : t -> Spec.Sub_zkapp.Stable.Latest.t option
```

Comment:
```suggestion
val next_job_spec : t -> Spec.Sub_zkapp.Stable.Latest.t option
```

And let's add a doc saying that it extracts another job spec from the structure (i.e. modifies the structure). That once job spec is completed, it's expected to be submitted back to structure with `submit_proof`.